scripts/coverage_report_jenkins.py

```diff
@@ -123,6 +123,45 @@ def get_build_numbers(client, job_name):
     return [build['number'] for build in client.get_job_info(job_name)['builds']]
 
 
+def gen_project_key(name, version):
+    """Generate project name based on Central CI rules
+
+    The Central CI docs found here:
+
+        https://docs.engineering.redhat.com/display/CentralCI/Code+Quality+Management
+
+    Document that a project key should take the form of:
+
+        <project-name>_<major_version>_<minor_version>_<language>_<coverage|static|full-analysis>
+
+    So given the name CFME and version 5.9.0.21, and that CFME is in ruby and we are
+    gathering coverage data, our project_key would be:
+
+        CFME_5_9_ruby_coverage
+
+    Args:
+        name:   application name
+        version:  A version like a.b.c.d where a major version and b is the minor version.
+            Actually minimally just need a.b, but any components after a.b are fine.
+    Returns:
+        a valid Central CI project key for sonarqube.
+    """
+    # I'm on purpose allowing for any number of version components after 2
+    # in case the version string changes (but still has major and minor at
+    # at the beginning.
+    match = re.search('^(?P<major>\d+)\.(?P<minor>\d+)', version)
+    if not match:
+        raise Exception(
+            'Invalid version string given.  Expect #.#[... .#] received: {}'.format(version))
```

Comment:
I get what you're doing with the "Expect:" string here but it's also a bit hard to understand at a glance ;) What if you just said exactly what you're looking for with regex and give an example -- like:

Start of version string must match '(\d+).(\d+)', e.g. '5.9'